graphics/persistent_cache.cc

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "flutter/shell/common/persistent_cache.h"
+#include "flutter/graphics/persistent_cache.h"
```

Comment:
I'm not wild about having flutter/graphics have a dependency into flutter/shell (for version). I know version is essentially self-contained, but shell/* is exactly what we don't want embeddings built on embedder.h to have arbitrary dependencies into. That's the whole reason for this move, after all.

I think we need a clearer definition of what dependencies are allowable, and a directory structure that makes the line much crisper. (I really wish we had Chromium's tooling for whitelisting/blacklisting cross-directory dependencies set up.)

Reply:
I agree, I was contemplating moving `version` out of shell to flutter root. Do you think that would make things easier to reason about?

Reply:
I'm not sure how many things we want to move to the root; these two things are probably not going to be the last things we want to share.

I'm increasingly thinking we should instead have a new directory in shell, something like `shell/embedding_utils` (it's not a great name, but naming it is non-trivial, especially given the existing `common`, which now isn't), and make that the firewall, and put `graphics` and `version` in there, along with an explanatory README.

I'll workshop a name in the engine channel.

Reply:
@stuartmorgan given that this will likely take some thinking. I'd like to not let this PR go stale. We can always move this folder to the right home once we decide on it. I've also added a warning message in the gn file to not add additional deps.

Reply:
My concern is that this gives the appearance of being isolated from shell/ by moving out of shell, but since it actually does depend back into shell that's not actually the case.

Even if we don't think this is a final location, what's the argument for having the interim location be both top-level and having dependencies into shell?

Reply:
I will move this into `//flutter/shell/graphics/`

Reply:
What do you think of `//flutter/common/graphics` so we don't inflate the root? Maybe `version` can also go there as both of them don't seem to rely on `shell` to exist.